server/apps/archived/archived.py

```diff
@@ -78,6 +92,129 @@ def on_create(self, docs):
                 if is_takes_package:
                     doc[SEQUENCE] = len(package_service.get_item_refs(doc))
 
+    def on_delete(self, doc):
+        """
+        Overriding to validate the item being killed is actually eligible for kill. Validates the following:
+            1. Is item of type Text?
+            2. Is item a Broadcast Script?
+            3. Does item acts as a Master Story for any of the existing broadcasts?
+            4. Is item available in production or part of a normal package?
+            5. Is the associated Digital Story is available in production or part of normal package?
+            6. If item is a Take then is any take available in production or part of normal package?
+        :param doc: represents the article in archived collection
+        :type doc: dict
+        :raises SuperdeskApiError.badRequestError() if any of the above validation conditions fail.
+        """
+
+        bad_req_error = SuperdeskApiError.badRequestError
+
+        id_field = doc[config.ID_FIELD]
+        item_id = doc['item_id']
+
+        doc['item_id'] = id_field
+        doc[config.ID_FIELD] = item_id
+
+        if doc[ITEM_TYPE] != CONTENT_TYPE.TEXT:
+            raise bad_req_error(message='Only Text articles are allowed to Kill in Archived repo')
+
+        if is_genre(doc, BROADCAST_GENRE):
+            raise bad_req_error(message="Killing of Broadcast Items isn't allowed in Archived repo")
+
+        if get_resource_service('archive_broadcast').get_broadcast_items_from_master_story(doc, True):
+            raise bad_req_error(message="Can't kill as this article acts as a Master Story for existing broadcast(s)")
+
+        if get_resource_service(ARCHIVE).find_one(req=None, _id=doc[GUID_FIELD]):
+            raise bad_req_error(message="Can't Kill as article is still available in production")
+
+        if is_item_in_package(doc):
+            raise bad_req_error(message="Can't kill as article is part of a Package")
+
+        takes_package_service = TakesPackageService()
+        takes_package_id = takes_package_service.get_take_package_id(doc)
+        if takes_package_id:
+            if get_resource_service(ARCHIVE).find_one(req=None, _id=takes_package_id):
```

Comment:
Agreed. But if the process fails in between while moving the takes package then takes package will be in production. Although the takes package eventually gets moved to archived, if user tries to kill a take before the process kicks in then this check will prevent.